bokeh/core/has_props.py

```diff
@@ -529,21 +529,23 @@ def query_properties_with_values(self, query, include_defaults=True):
             dict : mapping of property names and values for matching properties
 
         '''
+        themed_keys = set()
         result = dict()
         if include_defaults:
             keys = self.properties()
         else:
             keys = set(self._property_values.keys())
             if self.themed_values():
-                keys |= set(self.themed_values().keys())
+                themed_keys = set(self.themed_values().keys())
+                keys |= themed_keys
 
         for key in keys:
-            prop = self.lookup(key)
-            if not query(prop):
+            descriptor = self.lookup(key)
+            if not query(descriptor):
                 continue
 
-            value = prop.serializable_value(self)
-            if not include_defaults:
+            value = descriptor.serializable_value(self)
+            if not include_defaults and key not in themed_keys:
```

Comment:
OK I think I understand what's happening here now and agree with this piece; _unmodified_default_value could be a theme value or a default value, and if a key was themed, we can assume it was a theme value.

It would be sort of verbose but perhaps clearer if we changed that to `_unmodified_default_or_theme_value` I don't know. Though there's also the method name `instance_default` which is maybe similarly misleading. I guess for some of the code "default" means theme or default, but for other parts (like `include_defaults` here) it means strictly default. We care about strictly default for purposes of what to ship to JS. We care about theme OR default for purposes of whether the value in `_property_values` should override because application code has set it.